apps/speed.c

```diff
@@ -521,33 +519,29 @@ static EVP_MD *obtain_md(const char *name, int *fetched)
 
 static int have_md(const char *name)
 {
-    int fetched = 0;
     int ret = 0;
-    EVP_MD *md = obtain_md(name, &fetched);
+    EVP_MD *md = obtain_md(name);
 
     if (md != NULL) {
         EVP_MD_CTX *ctx = EVP_MD_CTX_new();
 
         if (ctx != NULL && EVP_DigestInit(ctx, md) > 0)
             ret = 1;
         EVP_MD_CTX_free(ctx);
-        if (fetched)
-            EVP_MD_free(md);
+        EVP_MD_free(md);
     }
     return ret;
 }
 
-static EVP_CIPHER *obtain_cipher(const char *name, int *fetched)
+static EVP_CIPHER *obtain_cipher(const char *name)
```

Comment:
Likewise, it looks to me like this can be dropped, and replaced with a call to `opt_cipher()`